client/src/api/iqdata/BlobClient.ts

```diff
@@ -3,34 +3,91 @@ import { convertToFloat32 } from '@/utils/FetchMoreDataSource';
 import { getBlobClient } from '@/api/utils/AzureBlob';
 import { SigMFMetadata } from '@/utils/sigmfMetadata';
 import { DataSource, IQDataSlice } from '@/api/Models';
+import { BlobClient as AzureBlobClient } from '@azure/storage-blob';
+import { groupContingousIndexes } from '@/utils/group';
 
 export class BlobClient implements IQDataClient {
   dataSources: Record<string, DataSource>;
 
   constructor(dataSources: Record<string, DataSource>) {
     this.dataSources = dataSources;
   }
+  async getIQDataBlocks(
+    meta: SigMFMetadata,
+    indexes: number[],
+    blockSize: number,
+    signal: AbortSignal
+  ): Promise<IQDataSlice[]> {
+    console.debug('getIQDataBlocks', indexes);
+    const contingousIndexes = groupContingousIndexes(indexes);
+    let { account, container, file_path } = meta.getOrigin();
+    // if filePath does not finish in .sigmf-data, add it
```

Comment:
Could there be any reason a file doesn't end with .sigmf-data because it isn't that, or would a check be unnecessary?

Reply:
It would not normally come with .sigmf-data because trhe way we normally display the filename it is without the extrension because the same file name it is used for thumbnail metadata and iqdata